doc/util/collect-syntax.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+
+"""
+collect syntax
+"""
+
+
+from __future__ import print_function
+import argparse
+import os
+import re
+import sys
+import stat
+import traceback
+
+
+# Regular expressions to identify productions within RST files.
+_match_syntax_begin = re.compile(r"\s*.. code-block:: syntax\s*\n")
+_match_syntax_end = re.compile(r"\S+")
+_match_production = re.compile(r" {3}(?P<ident>([a-zA-Z0-9._-]+)):.*")
+_match_expansion = re.compile(r" {3} +")
+_match_nonterminal = re.compile(r"\s*('[ a-zA-Z0-9-()]+')|(\\[a-zA-Z]+)|([a-zA-Z0-9-]+)")
+
+
+# Global variables...
+_productions = {}
+_do_list_productions = False
+_rstfiles = None
+_outdir = None
+_outfile = None
+_verbose = False
```

Comment:
It seems odd to me that all of these variables begin with `_`. Is there some Python convention I am not aware of?

Reply:
It's just a habit of indicating weak internal use that isn't really necessary in this script (I forced myself not to do it for functions). I normally don't use many global variables in Python, is there a convention we'd like to use for them? Or just named like anything else?

Reply:
I don't have any particular opinion on it.